packages/dds/merge-tree/src/test/obliterateOperations.ts

```diff
@@ -0,0 +1,215 @@
+	}
+	return startField ?? endField;
+};
+
+export const insertField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const chunkLength = random.integer(1, 10);
+	const numberText: string = (client.longClientId!.codePointAt(0)! % 10)
+		.toString()
+		.repeat(chunkLength);
+	if (posInField(client, opStart) === undefined) {
+		return client.insertTextLocal(opStart, `{${numberText}}`);
+	}
+};
+
+export const obliterateField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const fieldEndpoints = getFieldEndpoints(client, opStart, opEnd);
+
+	let endISP: InteriorSequencePlace | undefined;
+	if (fieldEndpoints !== undefined) {
+		const { startPos, endPos } = fieldEndpoints;
+		if (endPos >= client.getLength()) {
+			endISP = { pos: client.getLength() - 1, side: Side.After };
+		}
+		// Obliterate text bewteen the separators, but avoid the case where the obliterate range is zero length.
+		if (endPos - startPos > 1) {
+			const op = client.obliterateRangeLocal(
+				{ pos: startPos + 1, side: Side.Before },
+				endISP ?? { pos: endPos - 1, side: Side.After },
+			);
+			insertField(client, startPos + 1, endPos, random);
+			return op;
+		}
+	}
+	if (opEnd >= client.getLength()) {
+		endISP = { pos: client.getLength() - 1, side: Side.After };
+	}
+	return client.obliterateRangeLocal(
+		{ pos: opStart, side: Side.Before },
+		endISP ?? { pos: opEnd, side: Side.After },
+	);
+};
+
+export const insertAvoidField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const segs: ISegmentPrivate[] = [];
+	// gather all the segments at the pos, including removed segments
+	walkAllChildSegments(client.mergeTree.root, (seg) => {
+		const pos = client.getPosition(seg);
+		if (pos >= opStart) {
+			if (pos <= opStart) {
+				segs.push(seg);
+				return true;
+			}
+			return false;
+		}
+		return true;
+	});
+	if (segs.length > 0) {
+		const text = client.longClientId!.repeat(random.integer(1, 3));
+		const seg = random.pick(segs);
+		const movedOrRemoved = toRemovalInfo(seg) ?? toMoveInfo(seg);
+		const randomOffset = random.integer(0, seg.cachedLength - 1);
+		const initialPos = client.getPosition(seg) + randomOffset;
+		const endpoints = posInField(client, initialPos);
+
+		let offset = randomOffset;
+		if (endpoints !== undefined) {
+			const startSeg = client.getContainingSegment(endpoints.startPos);
+			assert(startSeg.offset !== undefined, "offset should be defined");
+			offset = startSeg.offset;
+		}
+
+		const lref = client.createLocalReferencePosition(
+			seg,
+			movedOrRemoved ? 0 : offset,
+			movedOrRemoved
+				? ReferenceType.SlideOnRemove
+				: random.pick([
+						ReferenceType.Simple,
+						ReferenceType.SlideOnRemove,
+						ReferenceType.Transient,
+					]),
+			undefined,
+		);
+
+		return client.insertAtReferencePositionLocal(lref, TextSegment.make(text));
```

Comment:
you don't need to replicate the complexity here or use insertAtReferencePositionLocal. just use insert test instead

Reply:
you could also do what i suggest below, and just call the existing function with corrected start and end